### PR TITLE
Calculate a first metric (monthly cost of housing) and display them in the UI

### PR DESCRIPTION
**Pull request by tahini**
This puts in place the mechanics of where to do calculations in the backend and how to display them in the frontend.

## Summary by CodeRabbit

* **New Features**
  * Results now group entries by address and show a formatted monthly housing cost per property; server auto-populates per-address monthly cost when Results are updated.

* **Tests**
  * Added comprehensive unit and server-side tests covering rent/mortgage scenarios, multi-address flows, edge cases, ordering, and error handling.

* **Chores**
  * Added English and French localization for new UI text.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by tahini** [comment]
Here's what it looks like for now. CSS experts are free and most welcome to make it prettier! with a 2 columns, one for each address. Ideally following the current group layout? otherwise, it would get more complicated I suppose, we would need group template, which is not supported in Evolution.

Capture d’écran du 2025-12-10 17-27-18

File: localisation/src/survey/calculations/index.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        // Add a fallback for zero mortgage
+        const monthlyMortgagePayment =
+                address.mortgage === 0
+                    ? 0
+                    : mortgageMonthlyPayment(
+                        address.mortgage,
+                        address.interestRate / 100, // Convert percentage to decimal
+                        amortizationPeriod * 12 // Convert years to months
+                    );
+        const taxes = typeof address.taxes === 'number' ? address.taxes / 12 : 0;
```
Comment (greenscientist) [inline localisation/src/survey/calculations/index.ts]: taxesMonthly
